### ISSUE TITLE
nil from recover function could not be compared properly

### ISSUE THREAD

**Issue by topxeq** [containous/yaegi#659]
The following program `sample.go` has an unexpected result:

```go
package main

import "fmt"

func div(a, b int) (result int) {
	defer func() {
		r := recover()

		fmt.Printf("r = %#v\n", r)

		if r != nil {
			result = 0
		}
	}()

	return a / b
}

func main() {
	println(div(30, 2))

}

```

Expected result:
```console
$ yaegi ./sample.go

r = <nil>
15
```

Got:
```console
$ yaegi ./sample.go
r = <nil>
0
```

So the nil value r in the defer function is not real "nil" value?

**Comment by topxeq** [comment]
Is it fixed? I still cannot get the correct result in the newest version.

**Comment by mvertes** [comment]
Indeed, this bug is not fixed by the change in #633. Sorry for that, it's our mistake. We reopen the issue.
